pkg/provider/pods.go

```diff
@@ -447,25 +448,38 @@ func (p *Pod) IsUsingSRIOVWithMTU() (bool, error) {
 
 	// For each CNCF network, get its network attachment definition and check
 	// whether its config's type is "sriov"
-	oc := clientsholder.GetClientsHolder()
 
 	for _, networkName := range cncfNetworkNames {
 		log.Debug("%s: Reviewing network-attachment definition %q", p, networkName)
-		nad, err := oc.CNCFNetworkingClient.NetworkAttachmentDefinitions(
-			p.Namespace).Get(context.TODO(), networkName, metav1.GetOptions{})
-		if err != nil {
-			return false, fmt.Errorf("failed to get NetworkAttachment %s: %v", networkName, err)
-		}
 
-		isSRIOV, err := isNetworkAttachmentDefinitionSRIOVConfigMTUSet(nad.Spec.Config)
-		if err != nil {
-			return false, fmt.Errorf("failed to know if network-attachment %s is sriov with MTU: %v",
-				networkName, err)
+		var foundNad nadClient.NetworkAttachmentDefinition
+
+		//nolint: gocritic
+		for _, nad := range env.NetworkAttachmentDefinitions {
```

Comment:
What's the exact purpose of this first loop? Is it checking if the NAD has the SRIOV type and the MTU defined? Because 1) MTU is still not checked here, and 2) foundNad is not really used. I suppose we should not run the next loop for sriovNetwork if foundNad is not empty, right?

Reply:
Yeah I was just playing around with looping through `env.NetworkAttachmentDefinitions` instead of doing the Get inside of the test itself.  I'll probably switch it back to the way it was.